pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiContributors.scala

```diff
@@ -0,0 +1,276 @@
+    *
+    */
+  def scribes(
+    xml: Elem,
+    workId: String): Result[Map[String, List[Contributor[Unminted]]]] =
+    (xml \\ "physDesc" \ "handDesc" \ "handNote").foldLeft(
+      Right(Map.empty): Result[Map[String, List[Contributor[Unminted]]]]
+    ) {
+      case (Left(err), _) => Left(err)
+      case (Right(scribesMap), node) =>
+        for {
+          contributor <- getScribeContributor(node)
+        } yield mapContributorToWorkId(workId, node, contributor, scribesMap)
+    }
+
+  /**
+    * Author nodes can be in 2 forms:
+    * <msItem xml:id="MS_Arabic_1-item1">
+    *  <author key="person_97166546">
+    *    <persName xml:lang="en">Avicenna, d. 980-1037
+    *    </persName>
+    *    <persName xml:lang="ar" type="original">ابو على الحسين ابن عبد الله ابن
+    *    سينا</persName>
+    *  </author>
+    *  or
+    * <msItem n="1" xml:id="MS_MSL_114_1">
+    *  <author key="person_84812936">Paul of Aegina</author>
+    *  So we must check for the existence of the internal persName nodes to decide
+    *  where to get the label and id from
+    */
+  private def getLabelAndId(n: Node) = (n \ "persName").toList match {
+    case Nil            => getFromAuthorNode(n)
+    case List(persNode) => getFromPersNode(n, persNode)
+    case list           => getFromOriginalPersNode(n, list)
+  }
+
+  private def getScribeContributor(n: Node) =
+    for {
+      maybeLabel <- parseScribeLabel(n)
+      contributor <- maybeLabel.map { label =>
+        createContributor(label, ContributionRole("scribe"))
+      }.sequence
+    } yield contributor
+
+  /**
+    * If the scribe refers to a nested part or item, it will have a locus tag with a target attribute like:
+    * <handNote scope="minor" scribe="Scribe_A"> <locus target="#Wellcome_Batak_36801_1 #Wellcome_Batak_36801_2" >a 2-62, b 2-7; b 8-22</locus> Gorak-gorahan, etc. Southern form of ta.</handNote>
+    *
+    * If the scribe refers to the wrapper work then it has no locus tag.
+    * Here we extract the id of the work the scribe refers to and add it to a map workId -> contributors
+    */
+  private def mapContributorToWorkId(
+    workId: String,
+    n: Node,
+    maybeContributor: Option[Contributor[Unminted]],
+    scribesMap: Map[String, List[Contributor[Unminted]]]) = {
+    maybeContributor match {
+      case None => scribesMap
+      case Some(contributor) =>
+        val workIds = extractNestedWorkIds(n)
+        workIds match {
+          case Nil =>
+            scribesMap + addIdToMap(workId, scribesMap, contributor)
+          case ids =>
+            scribesMap ++ addIdsToMap(scribesMap, contributor, ids)
+
+        }
+
+    }
+  }
+
+  /**
+    * The scribe name can be within the handNote tag directly or within a
+    * persName node with role="src" inside the handNote tag. If there is a persName tag,
+    * we pick that to construct the scribe name. Otherwise we use the text directly inside the handNote tag
+    */
+  private def parseScribeLabel(n: Node) = {
```

Comment:
To make this easier to follow/understand:

* Let's change the variable name to `handNote`, as that's the only type of node we're going to pass in here. This would be good to do throughout the helper functions – it'd be easier to map these helper functions back to the TEI XML.
* Pull out `n.attribute("scribe")` into a variable like `handNoteText`, so it more clearly maps to your comment.